base/MAPL_ISO8601_Time.F90

```diff
@@ -0,0 +1,720 @@
+!Allow 60 for seconds
+!No fractions other than milliseconds (explicit)
+!No weeks
+!Z required for time
+!No parts of day !Allow ordinal dates as input/output
+#include "MAPL_Exceptions.h"
+#include "MAPL_ErrLog.h"
+module MAPL_ISO8601_Time
+   use MAPL_KeywordEnforcerMod
+   use MAPL_ExceptionHandling
+   use ESMF
+   implicit none
+
+!   private
+   public :: DD
+   public :: TD
+   public :: TP
+   public :: TZ
+
+   !public :: string_to_integer_time
+   public :: string_to_integer_date
+   public :: string_to_esmf_time
+   !public :: string_to_esmf_timeinterval
+!   public :: operator(==)
+!
+!   interface operator(==)
+!      module procedure :: datefields_equal
+!      module procedure :: timefields_equal
+!      module procedure :: iso_date_equal
+!      module procedure :: iso_time_equal
+!      module procedure :: iso_datetime_equal
+!   end interface
+
+   ! parameters for processing date, time, and datetime strings
+   character, parameter :: DD = '-' ! Date Field Delimiter
+   character, parameter :: TD = ':' ! Time Field Delimiter
+   character, parameter :: TP = 'T' ! Time Prefix Character
+   character, parameter :: TZ = 'Z' ! Time Zone Character
+   character(len=*), parameter :: FDD = "(i4, 1x, i2, 1x, i2)" ! Format string for delimited date
+   character(len=*), parameter :: FUD = "(i4, i2, i2)" ! Format string for undelimited date
+   character(len=*), parameter :: FDT = "(1x, i2, 1x, i2, 1x, i2, 1x, i3, 1x)" ! Format string for delimited time
+   character(len=*), parameter :: FUT = "(1x, i2, i2, i2, 1x, i3, 1x)" ! Format string for undelimited time
+   character(len=*), parameter :: DATE_FORMAT = "(4i,a,2i,a,2i)"
+   character(len=*), parameter :: TIME_FORMAT = "(i2,a,2i,a,f)"
+   integer, parameter :: DIM_DF = 3 ! Number of fields in date
+   integer, parameter :: DIM_TF = 5 ! Number of fields in time
+
+   ! These are open lower(LB) and upper (UB) bounds of components of the date and time.
+   integer, parameter :: LB_YEAR = -1 
+   integer, parameter :: UB_YEAR = 10000
+   integer, parameter :: LB_MONTH = 0
+   integer, parameter :: UB_MONTH = 13
+   integer, parameter :: NUM_MONTHS = UB_MONTH - 1
+   integer, parameter :: LB_DAY = 0
+   integer, parameter :: LB_TIME = -1
+   integer, parameter :: LB_HOUR = LB_TIME
+   integer, parameter :: UB_HOUR = 24
+   integer, parameter :: LB_MINUTE = LB_TIME
+   integer, parameter :: UB_MINUTE = 60
+   integer, parameter :: LB_SECOND = LB_TIME
+   integer, parameter :: UB_SECOND = UB_MINUTE
+   integer, parameter :: LB_MILLISECOND = -1
+   integer, parameter :: UB_MILLISECOND = 1000
+
+   integer, parameter :: TIMEZONE_DEFAULT = 0
+
+   integer, parameter :: ID_WIDTH = 100
+   integer, parameter :: ID_DAY = 1
+   integer, parameter :: ID_MONTH = ID_WIDTH * ID_DAY
+   integer, parameter :: ID_YEAR = ID_WIDTH * ID_MONTH
+
+   type :: date_fields
```

Comment:
Preference is to have type names in CamelCase.    Also types are usually singular nouns, though more thought would be needed to come up with something that is not plural here.